scipy/spatial/ckdtree/src/ckdtree_count_neighbors.cxx

```diff
@@ -0,0 +1,223 @@
+#include <Python.h>
+#include "numpy/arrayobject.h"
+
+#include <cmath>
+#include <cstdlib>
+#include <cstring>
+
+#include <vector>
+#include <string>
+#include <sstream>
+#include <new>
+#include <typeinfo>
+#include <stdexcept>
+#include <ios>
+
+#define CKDTREE_METHODS_IMPL
+#include "ckdtree_cpp_decl.h"
+#include "ckdtree_cpp_methods.h"
+#include "ckdtree_cpp_exc.h"
+#include "ckdtree_cpp_rectangle.h"
+
+static void
+__count_neighbors_traverse(const ckdtree      *self,
```

Comment:
The C++ compilers do not complain about the double underscores, but I'll check. They are in the Cython code, but it is trivial to remove them.